src/blackout.c

```diff
@@ -0,0 +1,225 @@
+    env->msgFmt = MessageFormat_New(HEAP_ID_FIELD);
+
+    AddWindow(env->bgConfig, &env->window, &Blackout_WindowTemplate);
+
+    BufferPlayersName(env->msgFmt, 0, Save_PlayerData_GetProfileAddr(FieldSystem_GetSaveData(fieldSystem)));
+    if (fieldSystem->location->mapId == MAP_T20R0201) {
+        // {STRVAR_1 3, 0, 0} scurried back\nhome, protecting the exhausted\nand fainted Pokémon from further\nharm...
+        Blackout_PrintMessage(env, msg_0203_00004, 0, 0);
+    } else {
+        // {STRVAR_1 3, 0, 0} scurried to\na Pokémon Center, protecting\nthe exhausted and fainted\nPokémon from further harm...
+        Blackout_PrintMessage(env, msg_0203_00003, 0, 0);
+    }
+    CopyWindowToVram(&env->window);
+    TaskManager_Call(taskManager, Task_ShowPrintedBlackoutMessage, env);
+}
+
+static BOOL Task_ShowPrintedBlackoutMessage(TaskManager *taskManager) {
+    BlackoutScreenEnvironment *env = TaskManager_GetEnvironment(taskManager);
+    switch (env->state) {
+        case 0:
+            BeginNormalPaletteFade(3, 1, 43, RGB_WHITE, 8, 1, HEAP_ID_32);
+            G2_BlendNone();
+            env->state++;
+            break;
+        case 1:
+            if (IsPaletteFadeFinished()) {
+                env->state++;
+            }
+            break;
+        case 2:
+            if (gSystem.newKeys & PAD_BUTTON_A || gSystem.newKeys & PAD_BUTTON_B || gSystem.touchNew != 0) {
+                BeginNormalPaletteFade(0, 0, 0, RGB_BLACK, 8, 1, HEAP_ID_32);
+                env->state++;
+            }
+            break;
+        case 3:
+            if (IsPaletteFadeFinished()) {
+                FillWindowPixelBuffer(&env->window, 0);
+                env->state++;
+            }
+            break;
+        case 4:
+            ClearFrameAndWindow2(&env->window, FALSE);
+            RemoveWindow(&env->window);
+            MessageFormat_Delete(env->msgFmt);
+            DestroyMsgData(env->msgData);
+            FreeBgTilemapBuffer(env->bgConfig, GF_BG_LYR_MAIN_3);
+            FreeToHeap(env->bgConfig);
+            FreeToHeap(env);
+            return TRUE;
+    }
+
+    return FALSE;
+}
+
+static void Blackout_PrintMessage(BlackoutScreenEnvironment *environment, s32 msgNo, u8 x, u8 y) {
+    String *tmpStr = String_New(1024, HEAP_ID_FIELD);
+    String *finStr = String_New(1024, HEAP_ID_FIELD);
+
+    FillWindowPixelBuffer(&environment->window, 0);
+    ReadMsgDataIntoString(environment->msgData, msgNo, tmpStr);
+    StringExpandPlaceholders(environment->msgFmt, finStr, tmpStr);
+        
+    u32 width = FontID_String_GetWidthMultiline(0, finStr, 0);
+    x = (environment->window.width * 8 - width);
+    x /= 2;
+    x -= 4;
+
+    AddTextPrinterParameterized2(&environment->window, 0, finStr, x, y, TEXT_SPEED_NOTRANSFER, MAKE_TEXT_COLOR(1, 2, 0), NULL);
+
+    String_Delete(tmpStr);
+    String_Delete(finStr);
+}
+
+BOOL Task_BlackOut(TaskManager *taskManager) {
```

Comment:
Why is there inconsistent capitalization between Blackout from the above function and BlackOut right here?

Reply:
will address, which is preferable?

Reply:
I prefer Blackout but you should check to see what the message files use (one word vs two)